frontend/js/tests/user/import-listens/ImportListens.test.tsx

```diff
@@ -0,0 +1,164 @@
+    created: "2025-08-12T10:00:00Z",
+    metadata: {
+      filename: "listenbrainz.zip",
+      progress: "Import completed!",
+      status: "completed",
+    },
+    from_date: "2025-08-10T00:00:00Z",
+    to_date: "2025-08-12T00:00:00Z",
+  },
+];
+
+describe("ImportListensPage", () => {
+  let server: SetupServerApi;
+  let router: ReturnType<typeof createMemoryRouter>;
+
+  beforeAll(() => {
+    window.scrollTo = jest.fn();
+    window.HTMLElement.prototype.scrollIntoView = jest.fn();
+
+    server = setupServer(
+      http.post("/settings/import/", () => {
+        return HttpResponse.json({ user_has_email: true });
+      }),
+      http.get("/1/import-listens/list/", () => {
+        return HttpResponse.json(mockImports);
+      }),
+      http.post("/1/import-listens/cancel/1/", () => {
+        return HttpResponse.json({ success: true, id: 1 });
+      }),
+      http.post("/1/import-listens/1/", () => {
+        return HttpResponse.json(mockImports[0]);
+      })
+    );
+    server.listen();
+
+    router = createMemoryRouter(routes, {
+      initialEntries: ["/settings/import/"],
+    });
+  });
+
+  afterAll(() => {
+    server.close();
+  });
+
+  it("renders submission form correctly", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+    await waitFor(() => {
+      expect(screen.getByText(/start import from/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/end date for import/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/select Service/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText(/choose a File/i)).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByRole("button", { name: /import listens/i })).toBeInTheDocument();
+      
+    });
+  });
+  
+  it("renders imports with correct data", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText("August 11th, 2025")).toBeInTheDocument();
+      
+    });
+    await waitFor(() => {
+      expect(screen.getByText("-")).toBeInTheDocument();
+      
+    });
+    
+      
+    
+  });
+
+  it("calls cancel endpoint when cancel clicked", async () => {
+    renderWithProviders(
+      <RouterProvider router={router} />,
+      {},
+      { wrapper: ReactQueryWrapper },
+      false
+    );
+
+    await waitFor(() => {
+      expect(screen.getByText("spotify.zip")).toBeInTheDocument();
+    });
+    const cancelBtn = screen.getByRole("button", { name: "Cancel import" });
+    await user.click(cancelBtn);
```

Comment:
The setup is good, but there are no test assertions !
You should ensure that the server got hit with an API call to `POST /1/import-listens/cancel/1` , which if you look at the console when running the tests is not currently defined in the mock server.

The best way to do this with MSW is to have the mock server return a success response or a failure response, then check that the front-end shows the toast message you expect for that scenario.
That way you are not testing the internals of the component but rather the user interaction: locate button, click button, see success message.